ci/scripts/integration_hdfs.sh

```diff
@@ -22,20 +22,57 @@ set -e
 source_dir=${1}/cpp
 build_dir=${2}/cpp
 
-export CLASSPATH=`$HADOOP_HOME/bin/hadoop classpath --glob`
+export CLASSPATH=$($HADOOP_HOME/bin/hadoop classpath --glob)
 export HADOOP_CONF_DIR=$HADOOP_HOME/etc/hadoop
 export LIBHDFS3_CONF=$HADOOP_CONF_DIR/hdfs-site.xml
-export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$HADOOP_HOME/lib/native/
+export ARROW_LIBHDFS3_DIR=$CONDA_PREFIX/lib
+
+libhdfs_dir=$HADOOP_HOME/lib/native
+hadoop_home=$HADOOP_HOME
+
+function use_hadoop_home() {
+  unset ARROW_LIBHDFS_DIR
+  export HADOOP_HOME=$hadoop_home
+}
+
+function use_libhdfs_dir() {
+  unset HADOOP_HOME
+  export ARROW_LIBHDFS_DIR=$libhdfs_dir
+}
```

Comment:
@kou Could you explain the difference between the two? (use_hadoop_home vs. use_libhdfs_dir)
Are there two sources for libhdfs?

Reply:
We need to find `${HADOOP_HOME}/lib/native/libhdfs.so` when user uses `HADOOP_HOME`.
We need to find `${ARROW_LIBHDFS_DIR}/libhdfs.so` when user uses `ARROW_LIBHDFS_DIR`.

If `HADOOP_HOME` is defined, `ARROW_LIBHDFS_DIR` isn't used. So we need to test them separately.